lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -352,9 +346,16 @@ if Enum.all?(
             [
               link({:track_sync, track.stream_id})
               |> via_out(Pad.ref(:output, :audio))
-              |> to({:opus_decoder, track.id})
-              |> to({:aac_encoder, track.id})
-              |> to({:aac_parser, track.id})
+              |> then(
+                if audio_encoding == :OPUS,
+                  do:
+                    &(to(&1, {:opus_decoder, track.id}, Membrane.Opus.Decoder)
+                      |> to({:aac_encoder, track.id}, Membrane.AAC.FDK.Encoder)
+                      |> to({:aac_parser, track.id}, %Membrane.AAC.Parser{
+                        out_encapsulation: :none
+                      })),
+                  else: & &1
```

Comment:
I would move it to a variable e.g: `maybe_aac_transcoding` or something like this.